backend/GymBeam/QueryHandlers/ReservationQueryHandler.cs

```diff
@@ -0,0 +1,41 @@
+﻿using Domain;
+using GymBeam.Queries;
+using GymBeam.Response;
+using MediatR;
+
+namespace GymBeam.QueryHandlers
+{
+    public class ReservationQueryHandler : IRequestHandler<GetAllReservationsQuery, IEnumerable<ReservationResponse>>
+    {
+        private readonly IRepository _repository;
+
+        public ReservationQueryHandler(IRepository repository)
+        {
+            _repository = repository;
+        }
+
+        public Task<IEnumerable<ReservationResponse>> Handle(GetAllReservationsQuery request, CancellationToken cancellationToken)
+        {
+            var reservations = _repository.GetReservations();
+
+            if (reservations == null || !reservations.Any())
```

Comment:
|| !reservations.Any()
not needed if there is not any then select will return empty